### PR TITLE
Enable snapshot deployments

### PR DESCRIPTION
**Pull request by reinhapa**
Deploys a SNAPSHOT to [Sonatype](https://oss.sonatype.org/content/repositories/snapshots/org/codehaus/izpack/) after each successful build of the `master` branch.

### CODE REVIEW

**Comment by akuhtz** [comment]
@reinhapa I have no experience with github actions to deploy to sonatype. It looks too different from the gitlab ones that I currently use.
If it works I'm fine with it :-)

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -15,23 +15,43 @@ jobs:
...
+        run: mvn -B -U verify
+      - name: Running Maven GUI tests
+        run: |
+          sudo apt-get install fluxbox
+          export DISPLAY=':99.0'
+          Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+          fluxbox &> ~/fluxbox.log &
+          mvn -B verify -Pwith-gui-tests
+  deploy:
+    if: github.ref == 'refs/heads/main'
```
Comment (akuhtz) [inline .github/workflows/maven.yml:40]: Should be: ```if: contains( github.ref, 'master')```
See https://docs.github.com/en/actions/migrating-to-github-actions/migrating-from-gitlab-cicd-to-github-actions#condition-and-expression-syntax

**Review by akuhtz — COMMENTED** [review]
@reinhapa I guess the "if" in the deploy statement is wrong.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (reinhapa) [reply .github/workflows/maven.yml:40]: Fixed on master

**Pull request merged**